### ISSUE TITLE
NPE occurs when having an invalid entry in the web-ext

### ISSUE THREAD

**Issue by bpaskin-ibm** [OpenLiberty/open-liberty#6590]
When there is an invalid moduleName in web-ext within an application entry causes the application to fail to start and throws an NPE.  This is occurring in 19.0.0.1, but was ignored in 16.0.0.4.

```
   <application location="someear.ear" id="ear" type="ear">
        <web-ext moduleName="invalidname.war" context-root="/mycontext" />
   </application>
```

This would lead to ` E CWWKZ0002E: An exception occurred while starting the application ent-submission-tool-ear. The exception message was: com.ibm.ws.container.service.metadata.MetaDataException: java.lang.NullPointerException` in the messages.log and the following in the FFDC:

```
Caused by: java.lang.NullPointerException
        at com.ibm.ws.javaee.ddmodel.webext.WebExtAdapter.getConfigOverrides(WebExtAdapter.java:132)
        at com.ibm.ws.javaee.ddmodel.webext.WebExtAdapter.adapt(WebExtAdapter.java:61)
        at com.ibm.ws.javaee.ddmodel.webext.WebExtAdapter.adapt(WebExtAdapter.java:35)
        at com.ibm.ws.adaptable.module.internal.AdapterFactoryServiceImpl.adapt(AdapterFactoryServiceImpl.java:196)
        at com.ibm.ws.adaptable.module.internal.AdaptableContainerImpl.adapt(AdaptableContainerImpl.java:170)
        at com.ibm.ws.adaptable.module.internal.InterpretedContainerImpl.adapt(InterpretedContainerImpl.java:203)
        at com.ibm.ws.webcontainer.osgi.container.config.WebAppConfigurator.<init>(WebAppConfigurator.java:197)
        at com.ibm.ws.webcontainer.osgi.container.config.WebAppConfigurationAdapter.adapt(WebAppConfigurationAdapter.java:111)
        at com.ibm.ws.webcontainer.osgi.container.config.WebAppConfigurationAdapter.adapt(WebAppConfigurationAdapter.java:43)
        at com.ibm.ws.adaptable.module.internal.AdapterFactoryServiceImpl.adapt(AdapterFactoryServiceImpl.java:196)
        at com.ibm.ws.adaptable.module.internal.AdaptableContainerImpl.adapt(AdaptableContainerImpl.java:170)
        at com.ibm.ws.adaptable.module.internal.InterpretedContainerImpl.adapt(InterpretedContainerImpl.java:203)
        at com.ibm.ws.webcontainer.osgi.WebContainer.createModuleMetaData(WebContainer.java:833)
        ... 12 more
```

This should continued to be ignored, or at a minimum have a meaningful message.

Regards, Brian

**Comment by tbitonti** [comment]
Looks like this code is missing a null check on 'app':

```
          Application app = appInfo.getContainer().adapt(Application.class);
          for (Module m : app.getModules()) { // NPE
               moduleNames.add(stripExtension(m.getModulePath()));
          }
          configuredModuleNames.removeAll(moduleNames);
```

**Comment by tbitonti** [comment]
Fixed the NPE, but there is another problem.
The code generates errors correctly only when an application descriptor is present.  When no application descriptor is present, any time a web module has no web extension configuration extension, all web module configuration extensions which do not map to web modules explicitly listed in the application descriptor will be listed as invalid.

The conditions of an extraneous error are probably uncommon, but not too uncommon.  An extraneous error will show up if:
(1) There is no application.xml;
(2) There are at least two web modules;
(3) One of the web modules has a web module extension configuration override;
(4) The other web module does not have an override.

**Comment by tbitonti** [comment]
From developer conversations: 16.0.0.4 did not yet support configuration overrides.

**Comment by tbitonti** [comment]
Fixed by pull request #6605
